components/region_cache_memory_engine/src/engine.rs

```diff
@@ -251,6 +252,13 @@ pub struct RangeCacheMemoryEngine {
 
 impl RangeCacheMemoryEngine {
     pub fn new(config: Arc<VersionTrack<RangeCacheEngineConfig>>) -> Self {
+        RangeCacheMemoryEngine::with_region_info_provider(config, None)
+    }
+
+    pub fn with_region_info_provider(
```

Comment:
The final version of PR will have unit tests that have a mock region info provider for this.

Reply:
Will final version change Option<Arc<dyn RegionInfoProvider>> to Arc<dyn RegionInfoProvider>